ga4gh/datamodel/variants.py

```diff
@@ -474,17 +474,25 @@ def buildMetadata(
         ret = []
         header = varFile.header
         ret.append(buildMetadata(key="version", value=header.version))
-        for formatKey, value in header.formats.items():
-            if formatKey != "GT":
-                ret.append(buildMetadata(
-                    key="FORMAT.{}".format(value.name), type=value.type,
-                    number="{}".format(value.number)))
-                # NOTE: description is not currently implemented as a member
-                # of VariantMetadata in pysam/cbcf.pyx
-        for infoKey, value in header.info.items():
-            ret.append(buildMetadata(
-                key="INFO.{}".format(value.name), type=value.type,
-                number="{}".format(value.number)))
-            # NOTE: description is not currently implemented as a member
-            # of VariantMetadata in pysam/cbcf.pyx
+        formats = header.formats.items()
+        infos = header.info.items()
+        # TODO: currently ALT field is not implemented through pysam
+        # NOTE: contigs field is different between vcf files,
+        # so it's not included in metadata
+        # NOTE: filters in not included in metadata unless needed
+        keys = ["FORMAT", "INFO"]
+        for content in [formats, infos]:
+            for contentKey, value in content:
+                attrs = dict(value.header.attrs)
+                # TODO: refactor description at next pysam release
+                # since description will be implemented as a member of
+                # VariantMetadata
+                description = attrs.get('Description', '').strip('"')
+                key = "{0}.{1}".format(keys[0], value.name)
```

Comment:
Having `keys[0]` and `keys.pop(0)` for indexing in a loop is unintuitive.  Try something like this:
```
for i, content in enumerate([formats, infos]):
...
           key = "{0}.{1}".format(keys[i], value.name)
...
```